### PR TITLE
[DEV-9733] Adds additional logging for downloads

### PR DESCRIPTION
**Pull request by collinwr**
**Description:**
Adds some logging around downloads that are being marked as failed. 

**Technical details:**
- Simply logs the same message being stored in the db in the `fail_download()` method of `download_generation`. 
- Adds exception handling to the SQS step attempting to fail the job

**Requirements for PR merge:**

1. N/A Unit & integration tests updated
2. N/A API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. N/A Matview impact assessment completed
5. N/A Frontend impact assessment completed
6. N/A Data validation completed
7. N/A Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-9733](https://federal-spending-transparency.atlassian.net/browse/DEV-9733):
    - [x] Link to this Pull-Request
    - N/A Performance evaluation of affected (API | Script | Download)
    - N/A Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

**Review by kwhickey — COMMENTED** [review]
Just a few suggestions to look at

File: usaspending_api/download/management/commands/download_sqs_worker.py [hunk 1]
```diff
@@ -144,22 +144,33 @@ def _handle_queue_error(exc):
...
             logger=logger,
-            message=f"{type(exc).__name__} caught. Attempting to fail the DownloadJob",
+            message=f"{type(exc).__name__} caught. Attempting to fail the DownloadJob with id = {download_job_id}",
             job_type=JOB_TYPE,
             job_id=download_job_id,
             is_exception=True,
         )
         if download_job_id:
             stack_trace = "".join(traceback.format_exception(etype=type(exc), value=exc, tb=exc.__traceback__))
             status = "failed"
```
[the remark this answers is not in the document]
Comment (collinwr) [reply usaspending_api/download/management/commands/download_sqs_worker.py:154]: That string is ultimately used in `_update_download_job_status()` to lookup the `job_status_id`
> download_job.job_status_id = JOB_STATUS_DICT[status]

File: usaspending_api/download/filestreaming/download_generation.py [hunk 2]
```diff
@@ -718,6 +718,7 @@ def strip_file_extension(file_name):
 
 
 def fail_download(download_job, exception, message):
+    write_to_log(message=message, is_error=True, download_job=download_job)
```
[the remark this answers is not in the document]
Comment (collinwr) [reply usaspending_api/download/filestreaming/download_generation.py:721]: `log_job_message` seems to come from `sqs_job_logging.py` library. Most of the extra metadata fields relate to the parent/child processes, attempts, etc.

```
log_dict = {
        "job_type": job_type,
        "job_id": job_id,
        "proc_id": os.getpid(),
        "parent_proc_id": os.getppid(),
        "attempt": attempt,
        "message": message,
    }
```

`write_to_log` does pull out a few fields from the `download_job` and include them in the JSON response though. Using it seems to be the pattern in `download_generation.py`
```
def download_job_to_log_dict(download_job, log_dict=None):
    if not log_dict:
        log_dict = {}
    if download_job:
        log_dict["download_job_id"] = download_job.download_job_id
        log_dict["job_id"] = download_job.download_job_id  # to match log structure of generic queued jobs
        log_dict["file_name"] = download_job.file_name
        log_dict["json_request"] = download_job.json_request
        if download_job.error_message:
            log_dict["error_message"] = download_job.error_message
    return log_dict
```

**Review by kwhickey — APPROVED** [review]
Looks good

**Pull request merged**